File: src/main/java/italo/sisrest/model/Pedido.java
package italo.sisrest.model;

import java.util.List;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
@Entity
@Table(name="pedido")
public class Pedido {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private int mesa;

    private boolean atendido;

    @OneToMany(mappedBy = "pedido", orphanRemoval = true, cascade = CascadeType.ALL)
    private List<PedidoItem> items;

    @OneToOne(mappedBy = "pedido", cascade = CascadeType.REMOVE)
    private UltimoPedidoAtendido ultimoPedidoAtendido;

}

File: frontend/src/models/dtos/Pedido.ts
import { PedidoItem, SavePedidoItem } from "./PedidoItem";

export interface Pedido {
    id : number;
    mesa : number;
    atendido : boolean;
    items : PedidoItem[];
}

export interface SavePedido {
    mesa : number;
    items: SavePedidoItem[];
}